### DISCUSSION TITLE
Why are static hard coded IP used and can we make it dynamic?

### DISCUSSION THREAD

**Discussion by mohnishsiyodia** [vpnhood/VpnHood#427] [Q&A] [answered]
I have been using VpnHoodClient.Connect() to establish a vpn connection. In the Connect() method there is another method called as StartCapture().

The StartCapture() method is using a hardcoded Ip address in builder for VPN service. Refer to the snapshot attached.

![Screenshot 2023-11-29 120715](https://github.com/vpnhood/VpnHood/assets/53507901/e0d8781a-3747-4c53-b0b3-f3c3ff65d0e1)

Also the Ip address is different in different versions, why?

v2.1.276 - 192.168.0.100 
v3.1.436 - 192.168.199.188

Instead of adding a static Ip for the builder can we make it more dynamically?

**Comment by trudyhood** [comment]
I am still trying to figure it out. As far as I know, VpnService requires an IP range for its virtual adapter. We changed to 192.168.199.188 to make it less likely to conflict with any other service. Let me know if you know the best practice so I can change it.

**Comment by robsiera** [comment]
Oh, you are using google DNS internally? Google, the company that made us aware that privacy is a valuable thing? They are probably tracking all those DNS requests?
Might I ask to consider using a different default DNS? 
Of course, which one it probably a difficult question. I was convinced that Quad9 was the best option, but yesterday I discovered that they depend heavily on [PCH.net](https://en.wikipedia.org/wiki/Packet_Clearing_House) which is sponsored by Soros. So I switched to [Mullvad](https://mullvad.net/en/help/dns-over-https-and-dns-over-tls) but only after some common sense thinking. I wished I had found an security/privacy expert to help me choose. 
<html>
<body>

dns.mullvad.net | 194.242.2.2 | 2a07:e340::2

</body>
</html>

Anyway, just some thoughts. Love you work!

**Comment by trudyhood** [comment, marked as answer]
I appreciate your consideration. We will make the feature configurable, as discussed in this issue: https://github.com/vpnhood/VpnHood/issues/458

The default DNS server will be set to Google, as most clients prioritize reliability. Additionally, the DNS server within the VPN will only see the server IP and will not be aware of individual client requests. However, for those who prefer a different option, there will be the flexibility to change this setting on the server.

We are particularly grateful that you are reporting any consideration in the source code. Your engagement and feedback are invaluable to us, and we like to hear your thoughts and any considerations you might have.
